src/sidebar/components/user-menu.js

```diff
@@ -0,0 +1,86 @@
+'use strict';
+
+const { createElement } = require('preact');
+const propTypes = require('prop-types');
+
+const bridgeEvents = require('../../shared/bridge-events');
+const { isThirdPartyUser } = require('../util/account-id');
+const serviceConfig = require('../service-config');
+const { withServices } = require('../util/service-context');
+
+const Menu = require('./menu');
+const MenuSection = require('./menu-section');
+const MenuItem = require('./menu-item');
+
+/**
+ * A menu with user and account links.
+ *
+ * This menu will contain different items depending on service configuration,
+ * context and whether the user is first- or third-party.
+ */
+function UserMenu({ auth, bridge, onLogout, serviceUrl, settings }) {
+  const isThirdParty = isThirdPartyUser(auth.userid, settings.authDomain);
+  const service = serviceConfig(settings);
+
+  const serviceSupports = feature => service && !!service[feature];
+
+  const isSelectableProfile =
+    !isThirdParty || serviceSupports('onProfileRequestProvided');
+  const isLogoutEnabled =
+    !isThirdParty || serviceSupports('onLogoutRequestProvided');
+
+  const onProfileSelected = () =>
+    isThirdParty && bridge.call(bridgeEvents.PROFILE_REQUESTED);
+
+  // Generate the correct props for the profile <MenuItem> component
+  const profileItemProps = (() => {
+    const props = {};
+    if (isSelectableProfile) {
+      if (!isThirdParty) {
+        props.href = serviceUrl('user', { user: auth.username });
+      }
+      props.onClick = onProfileSelected;
+    } else {
+      props.isDisabled = true;
+    }
+    return props;
+  })();
```

Comment:
`isDisabled` here is just `!isSelectableProfile` so you could potentially assign that statically below. As I mentioned above, I find it useful when the output of a component get be fairly easily discerned just from looking at the returned JSX expression.